File: services/review-service/src/main/java/iuh/fit/se/dto/user/Address.java
package iuh.fit.se.dto.user;

import org.springframework.data.annotation.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = lombok.AccessLevel.PROTECTED)
@EqualsAndHashCode(onlyExplicitlyIncluded = true) // Quan trọng cho Set
public class Address {
	@Id
	@EqualsAndHashCode.Include
	String addressId;
	String recipientName;
	String recipientPhone;
	String recipientAddress;
	String addressType;

}

File: services/userService/models/address.js
module.exports = class Address {
    constructor(addressId, recipientName, recipientPhone, recipientAddress, addressType) {
      this.addressId = addressId;                   // String
      this.recipientName = recipientName;           // String
      this.recipientPhone = recipientPhone;         // String
      this.recipientAddress = recipientAddress;     // String
      this.addressType = addressType;               // int (ví dụ: 0 = nhà riêng, 1 = công ty)
    }
  };
  